feat: configure network rule set

musifalsk opened the pull request:
Added a conditional block  network_rule_set to the azurerm_servicebus_namespace resource.
and required variables.

musifalsk committed:
added block network_rule_set

musifalsk committed:
changes

musifalsk committed:
servicebus-queue

musifalsk committed:
modified servicebus module

musifalsk committed:
modified servicebus module

musifalsk committed:
modified servicebus module

musifalsk committed:
modified servicebus module

musifalsk committed:
modified servicebus module

musifalsk committed:
modified servicebus module

musifalsk committed:
modified servicebus module

musifalsk committed:
testing depends_on meta arg

musifalsk committed:
testing depends_on meta arg

musifalsk committed:
testing depends_on meta arg

musifalsk committed:
added networking block to config

musifalsk committed:
adjustments

musifalsk committed:
adjustments

hknutsen commented:
Updated PR title according to [contributing guidelines](https://github.com/equinor/terraform-baseline/blob/main/CONTRIBUTING.md).

hknutsen committed:
refactor: update network configuration (#52)

* chore: update required Azure provider version

Support for the `network_rule_set` block for resource `azurerm_servicebus_namespace` was added in Azure provider v3.71.

* refactor: rename variables

* refactor: remove variable for enabling network rules

* refactor: remove duplicate variable

* Fix configuration of Premium only features

* Deny public network access by default

* Dynamically configure network rules default action

* Rename variable

* Set optional object property

* Reorder blocks

hknutsen committed:
Merge branch 'main' into update-service-bus-module-networking

hknutsen committed:
Update main.tf
